### PR TITLE
GitHub Actions cross

### PR DESCRIPTION
**Pull request by crazy-max**
* Switch To GitHub Actions for cross target
  * also store artifacts temporarly (90d by default in repo settings) in the workflow (can remove that if you want)
  * and build dynbinary cross
  * ~use GitHub Action cache backend~
* Cleanup bake definition that was specific to CircleCI
* Restore `binary` target in `Makefile` for non-sandboxed build
* Fix README with proper lint command using bake

Run: https://github.com/docker/cli/actions/runs/1103549570

image

Signed-off-by: CrazyMax <crazy-max@users.noreply.github.com>

### CODE REVIEW

File: docker-bake.hcl [hunk 1]
```diff
@@ -32,34 +32,16 @@ target "dynbinary" {
...
-variable "GROUP_INDEX" {
-    default = "0"
-}
-
-function "platforms" {
-    params = [USE_GLIBC]
-    result = concat(["linux/amd64", "linux/386", "linux/arm64", "linux/arm", "linux/ppc64le", "linux/s390x", "darwin/amd64", "darwin/arm64", "windows/amd64", "windows/arm", "windows/386"], USE_GLIBC!=""?[]:["windows/arm64"])
-}
-
-function "glen" {
```
Comment (thaJeztah) [inline docker-bake.hcl:48]: n00b question; 😅 these are all no longer needed? Why were they needed before?

Comment (crazy-max) [reply docker-bake.hcl:48]: They were needed for CircleCI `cross.parallelism` which provides [`CIRCLE_NODE_INDEX` and `CIRCLE_NODE_TOTAL` env vars](https://circleci.com/docs/2.0/env-vars/#built-in-environment-variables) to handle efficient parallelism.

Comment (thaJeztah) [reply docker-bake.hcl:48]: Ah! Gotcha

**Comment by crazy-max** [comment]
PTAL @thaJeztah @silvin-lubecki :)

**Comment by crazy-max** [comment]
Removed GHA cache backend as we have issues with it.

**Comment by crazy-max** [comment]
As discussed with @thaJeztah I restored the `binary` target to allow non-sandboxed build.

**Review by thaJeztah — APPROVED** [review]
LGTM

**Comment by thaJeztah** [comment]
@silvin-lubecki PTAL

**Review by silvin-lubecki — APPROVED** [review]
LGTM

**Pull request merged**